pkg/gatherers/workloads/gather_workloads_info.go

```diff
@@ -225,15 +241,15 @@ func podCanBeIgnored(pod *corev1.Pod) bool {
 		len(pod.Status.ContainerStatuses) != len(pod.Spec.Containers)
 }
 
-func calculatePodShape(h hash.Hash, pod *corev1.Pod) (workloadPodShape, bool) {
+func calculatePodShape(h hash.Hash, pod *corev1.Pod, workloadInfo workloadRuntimes) (workloadPodShape, bool) {
 	var podShape workloadPodShape
 	var ok bool
-	podShape.InitContainers, ok = calculateWorkloadContainerShapes(h, pod.Spec.InitContainers, pod.Status.InitContainerStatuses)
+	podShape.InitContainers, ok = calculateWorkloadContainerShapes(h, pod.ObjectMeta, pod.Spec.InitContainers, pod.Status.InitContainerStatuses, workloadInfo)
 	if !ok {
 		return workloadPodShape{}, false
 	}
 
-	podShape.Containers, ok = calculateWorkloadContainerShapes(h, pod.Spec.Containers, pod.Status.ContainerStatuses)
+	podShape.Containers, ok = calculateWorkloadContainerShapes(h, pod.ObjectMeta, pod.Spec.Containers, pod.Status.ContainerStatuses, workloadInfo)
```

Comment:
Maybe it's time to pass the complete `Pod` reference here? Just suggestion.....

Reply:
that'll be cleaner. I'll do it

Reply:
@tremes Do you mean passing the pod instead of the `pod.ObjectMeta` (the method is called twice for the Containers and the imiticontainers)

Reply:
ohh right it needs to distinguish the containers and initContainers....humm so perhaps easier to keep it as it is :thinking: